stubs/pika/pika/adapters/blocking_connection.pyi

```diff
@@ -192,22 +199,24 @@ class BlockingChannel:
     def basic_ack(self, delivery_tag: int = ..., multiple: bool = ...) -> None: ...
     def basic_nack(self, delivery_tag: int = ..., multiple: bool = ..., requeue: bool = ...) -> None: ...
     def basic_get(self, queue, auto_ack: bool = ...): ...
-    def basic_publish(self, exchange, routing_key, body, properties: Incomplete | None = ..., mandatory: bool = ...) -> None: ...
+    def basic_publish(
+        self, exchange: str, routing_key: str, body: str | bytes, properties: BasicProperties | None = ..., mandatory: bool = ...
```

Comment:
The `body` parameter is explicitly encoded in the implementation if it's a `str`: https://github.com/pika/pika/blob/e1f7c83ad2279ca98d33151e59955262461dd509/pika/channel.py#L424

Generally, the `str`/`bytes` annotations in the docstrings are not terribly accurate, since many of them still stem from Python 2 times.

Reply:
> Generally, the `str`/`bytes` annotations in the docstrings are not terribly accurate, since many of them still stem from Python 2 times.

Big yikes!